packages/react-component-library/src/components/Select/Select.tsx

```diff
@@ -72,8 +72,16 @@ export const Select = ({
       hasSelectedItem={!isNil(selectedItem)}
       id={id}
       inputProps={{
-        onFocus: onInputFocusHandler,
+        ...getToggleButtonProps({
+          onFocus: onInputFocusHandler,
+          onMouseDown: (event) => {
+            // This is crucial for correct focus and click behavior
+            // Without this, Downshift internal state gets mixed up
+            event.preventDefault()
+          },
+        }),
         ref: inputRef,
+        'aria-expanded': undefined,
```

Comment:
This gets applied by `getToggleButtonProps` so we need to nullify it.

![Screenshot 2024-10-14 at 16 48 32](https://github.com/user-attachments/assets/3d53fe45-62b7-4835-91cc-b7cc7c051a86)

Reply:
Hmm yeah `getToggleButtonProps()` is being on two elements, maybe that's not right but hunting down the Downshift v6 docs isn't straightforward :smile: (Plus the WAI-ARIA spec has changed since v6...)

Reply:
Another one we need to upgrade at some point - both this and DatePicker are crying out for it.